### PR TITLE
Allow inheritance for addActionToClass

### PR DESCRIPTION
**Pull request by PabstMirror**
Handles the mess of adding actions to all units.
E.G. can add class actions to `CaManBase`.

### CODE REVIEW

**Comment by commy2** [comment]
This is badly needed, thanks!

File: addons/interact_menu/functions/fnc_addActionToClass.sqf [hunk 1]
```diff
@@ -8,20 +8,37 @@
  * 1: Type of action, 0 for actions, 1 for self-actions <NUMBER>
  * 2: Parent path of the new action <ARRAY>
  * 3: Action <ARRAY>
+ * 4: Use Inheritance (Default: False) <BOOL><OPTIONAL>
```
Comment (jonpas) [inline addons/interact_menu/functions/fnc_addActionToClass.sqf]: Standardly we do this like:
` * 4: Use Inheritance <BOOL> (default: false)`

**Comment by jonpas** [comment]
No more `GVAR(initializedClasses)`! :tada:

File: addons/interact_menu/functions/fnc_addActionToClass.sqf [hunk 2]
```diff
@@ -8,20 +8,37 @@
...
  *
  * Return Value:
  * The entry full path, which can be used to remove the entry, or add children entries <ARRAY>.
  *
  * Example:
  * [typeOf cursorTarget, 0, ["ACE_TapShoulderRight"],VulcanPinchAction] call ace_interact_menu_fnc_addActionToClass;
  *
- * Public: No
+ * Public: Yes
```
Comment (commy2) [inline addons/interact_menu/functions/fnc_addActionToClass.sqf]: should set `FUNC(createAction)` public=yes then too.

File: addons/interact_menu/functions/fnc_addActionToClass.sqf (hunk 1 above)
Comment (commy2) [inline addons/interact_menu/functions/fnc_addActionToClass.sqf]: `<OPTIONAL>` looks strange
`(optional, default: false) <BOOL(EAN)>` ?
Maybe it's just me.

**Comment by commy2** [comment]
>No more GVAR(initializedClasses)! :tada:

The PR should probably convert them to use this. Will go through stuff after I get home tomorrow.

**Comment by PabstMirror** [comment]
Dragging, Repair, Cargo, all do something similar but they are also reading configs, so it won't exactly be the same.

**Comment by jonpas** [comment]
Ah, that's very true, so dies Sitting for example. Not sure if it would be better to leave it like this or add the config reading in condition rather though.

File: docs/wiki/framework/interactionMenu-framework.md [hunk 3]
```diff
@@ -97,8 +97,10 @@ Important: `ace_common_fnc_canInteractWith` is not automatically checked and nee
  * 1: Type of action, 0 for actions, 1 for self-actions <NUMBER>
  * 2: Parent path of the new action <ARRAY>
  * 3: Action <ARRAY>
+ * 4: Use Inheritance (Default: False) <BOOL><OPTIONAL>
```
Comment (jonpas) [inline docs/wiki/framework/interactionMenu-framework.md]: Same here: ` * 4: Use Inheritance <BOOL> (default: false)`

**Comment by esteldunedain** [comment]
For performance reasons I think it's best to NOT use inherited actions for most modules. I think it's better to take that extra time at mission start to only add the actions to classes that need them instead of mass adding them to all classes and evaluating false condition every time player tries to interact near them.

File: docs/wiki/framework/interactionMenu-framework.md [hunk 4]
```diff
@@ -136,6 +138,18 @@ _action = ["Open RDF","Radio Direction Finder","pabst\RDF.jpg",_statement,_condi
 [(typeOf _unit), 1, ["ACE_SelfActions"], _action] call ace_interact_menu_fnc_addActionToClass;
 ```
 
+Using addActionToClass inheritance:
```
Comment (jonpas) [inline docs/wiki/framework/interactionMenu-framework.md]: `addActionToClass` (code)

**Pull request merged**